js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -0,0 +1,292 @@
+import {ApolloClient, useApolloClient} from '@apollo/client';
+import React from 'react';
+
+import {assertUnreachable} from '../app/Util';
+import {LiveDataForNode, buildLiveDataForNode} from '../asset-graph/Utils';
+import {
+  AssetGraphLiveQuery,
+  AssetGraphLiveQueryVariables,
+  AssetLatestInfoFragment,
+  AssetNodeLiveFragment,
+} from '../asset-graph/types/useLiveDataForAssetKeys.types';
+import {
+  ASSETS_GRAPH_LIVE_QUERY,
+  ASSET_LATEST_INFO_FRAGMENT,
+  ASSET_NODE_LIVE_FRAGMENT,
+} from '../asset-graph/useLiveDataForAssetKeys';
+import {AssetKeyInput} from '../graphql/types';
+import {isDocumentVisible, useDocumentVisibility} from '../hooks/useDocumentVisibility';
+
+const _assetKeyListeners: Record<string, Array<DataForNodeListener>> = {};
+
+export function useAssetNodeLiveData(assetKeys: AssetKeyInput[]) {
+  const context = React.useContext(AssetLiveDataContext);
+  return context.useAssetNodeLiveData(assetKeys);
+}
+
+function _getAssetFromCache(client: ApolloClient<any>, uniqueId: string) {
+  const cachedAssetData = client.readFragment<AssetNodeLiveFragment>({
+    fragment: ASSET_NODE_LIVE_FRAGMENT,
+    fragmentName: 'AssetNodeLiveFragment',
+    id: `assetNodeLiveFragment-${uniqueId}`,
+  });
+  const cachedLatestInfo = client.readFragment<AssetLatestInfoFragment>({
+    fragment: ASSET_LATEST_INFO_FRAGMENT,
+    fragmentName: 'AssetLatestInfoFragment',
+    id: `assetLatestInfoFragment-${uniqueId}`,
+  });
+  if (cachedAssetData && cachedLatestInfo) {
+    return {cachedAssetData, cachedLatestInfo};
+  } else {
+    return null;
+  }
+}
+
+async function _queryAssetKeys(client: ApolloClient<any>, assetKeys: AssetKeyInput[]) {
+  const {data} = await client.query<AssetGraphLiveQuery, AssetGraphLiveQueryVariables>({
+    query: ASSETS_GRAPH_LIVE_QUERY,
+    fetchPolicy: 'network-only',
+    variables: {
+      assetKeys,
+    },
+  });
+  data.assetNodes.forEach((assetNode) => {
+    _assetKeyToUniqueId[JSON.stringify(assetNode.assetKey.path)] = assetNode.id;
+    client.writeFragment({
+      fragment: ASSET_NODE_LIVE_FRAGMENT,
+      fragmentName: 'AssetNodeLiveFragment',
+      id: `assetNodeLiveFragment-${assetNode.id}`,
+      data: assetNode,
+    });
+  });
+  data.assetsLatestInfo.forEach((assetLatestInfo) => {
+    _assetKeyToUniqueId[JSON.stringify(assetLatestInfo.assetKey.path)] = assetLatestInfo.id;
+    client.writeFragment({
+      fragment: ASSET_LATEST_INFO_FRAGMENT,
+      fragmentName: 'AssetLatestInfoFragment',
+      id: `assetLatestInfoFragment-${assetLatestInfo.id}`,
+      data: assetLatestInfo,
+    });
+  });
+}
+
+// How many assets to fetch at once
+const BATCH_SIZE = 50;
+
+// Milliseconds we wait until sending a batched query
+const BATCHING_INTERVAL = 100;
+
+export const SUBSCRIPTION_IDLE_POLL_RATE = 30 * 1000;
+// const SUBSCRIPTION_MAX_POLL_RATE = 2 * 1000;
```

Comment:
It seems like this implementation removes the run log subscriptions and you just get live data refreshing at a 30s interval all the time? Right now the live data is refreshed immediately if a materialization or observation event arrives which is pretty important for demo-style usage of Dagster where you click "Materialize" and then see things on the graph refresh as you wait.  [and wait <30s]

Reply:
Yeah I will add that in a follow-up. I didn't want to make too many changes in this PR and that change seemed a little bit more involved to add.